io.go

```diff
@@ -44,6 +47,38 @@ func (cr *CborReader) SetReader(r io.Reader) {
 	cr.r = GetPeeker(r)
 }
 
+func (cr *CborReader) ReadCid(scratchBuf []byte) (cid.Cid, error) {
+	maj, extra, err := cr.ReadHeader()
+	if err != nil {
+		if err == io.EOF {
+			err = io.ErrUnexpectedEOF
+		}
+		return cid.Undef, err
+	}
+
+	if maj != MajTag {
+		return cid.Undef, fmt.Errorf("expected cbor type 'tag' in input")
+	}
+
+	if extra != 42 {
+		return cid.Undef, fmt.Errorf("expected tag 42")
+	}
+
+	if extra > 512 {
+		return cid.Undef, fmt.Errorf("header size too big for a cid")
+	}
+
+	_, err = io.ReadAtLeast(cr, scratchBuf[:extra], int(extra))
```

Comment:
Can I guarantee that the scratch buffer is large enough for this?

Reply:
added a check